src/strategies/information.rs

```diff
@@ -892,12 +895,17 @@ impl InformationPlayerStrategy {
         }
 
         // if anything is totally useless, discard it
-        if public_useless_indices.len() > 1 {
-            let info = public_info.get_hat_sum(public_useless_indices.len() as u32, view);
+        if public_useless_indices.len() > 0 {
+            let info_amt = public_useless_indices.len() as u32;
+            if let Some(private_useless_index) = private_useless_index {
+                let num_states = info_amt + 2;
```

Comment:
hmm, i might be confused, but shouldn't it just always be 2?  why also use info_amt?

Reply:
Ohh yes I should probably have elaborated on that better, sorry!
So we'll have a `1/num_states` chance of discarding the privately known to be useless card, in which case we transmit `log2(num_states)` bits of information. All else being equal, we thus want to maximize `log2(num_states)/num_states`, which we get by setting `num_states = 3`.

But if `info_amt != 0`, all else is not equal, since if we don't choose to play the privately known useless card, we get to transmit `log2(info_amt)` bits of information. Therefore, we want to increase `num_states` so that we're less likely to play the privately known card (and learn more if we do).

Actually, writing this got me to plug this into Wolfram Alpha, and there's a nice optimum: I should have set `num_states` to some rounding of `e * info_amt`! I don't know why I didn't have that idea earlier.

Reply:
my (weak) intuition is actually that *always* playing the privately known useless card will do better than all of this, in the limit

advantages:
- you keep more publicly known discards, which lead to more info transmission later on (although maybe this mainly happens in endgame)
- when you discard a publicly known card, other players learn that you don't have a privately known dead card.  that's a lot of information in the (admittedly) rare case it happens.  the information is typically of the form "i have a red 4", though, which we currently can't make use of

disadvantage:
- when the card is discarded you dont transmit any additional question-based info